### ISSUE TITLE
default values

### ISSUE THREAD

**Issue by bokodi-dev** [unclead/yii2-multiple-input#99]
The default values are not working. Code sample:
```
<?= TabularInput::widget([
    'models' => $models,
    'columns' => [
        [
            'name' => 'call_id',
            'type' => TabularColumn::TYPE_HIDDEN_INPUT
        ],
        [
            'name' => 'subject',
            'title' => 'subject',
            'type' => TabularColumn::TYPE_TEXT_INPUT,
            'defaultValue' => 'default subject',
            'enableError' => true
        ],
        [
            'name' => 'comment',
            'title' => 'comment',
        ],
        [
            'name' => 'created_at',
            'title' => 'Created At',
            'type'  => DatePicker::className(),
            'defaultValue' => '2000-01-01',
            'options' => [
                'pluginOptions' => [
                    'format' => 'yyyy-mm-dd',
                    'todayHighlight' => true
                ]
            ]
        ]
    ]
]) ?>
```

Also it would be nice to have the ability to init the TabularInput widget with an empty array:
```
<?= TabularInput::widget([
    'models' => [],
    'columns' => [
```
throws error: You must specify "models"

**Comment by unclead** [comment]
> Also it would be nice to have the ability to init the TabularInput widget with an empty array:

you can pass to the widget an array wich contains one new model without any data. Currently widget is using the model for getting input name and id and it is impossible to do it without a model.

> The default values are not working.
have you tried find out a problem? i'm too busy now and don't have enough time to fix it right now
It would be great if you try to fix it by your own and send me a PR

**Comment by unclead** [comment]
closed via edff08e0169274dee63ecf7c62d3be224fd1e9d8

**Comment by bokodi-dev** [comment]
It's working with ``'type' => TabularColumn::TYPE_TEXT_INPUT,`` but the widgets are still not using the default values, like kartik switchinput or select2.

In BaseColump.php at line 476 the renderWidget method passes the model to the widget, so the widget uses the model's attribute instead of the default values:
```
$widgetOptions = [
    'model'     => $model, // here
    'attribute' => $this->name,
    'value'     => $value,
    'options'   => [
        'id' => $this->normalize($name),
        'name' => $name
    ]
];

...

$options = ArrayHelper::merge($options, $widgetOptions);

return $type::widget($options);
```

**Comment by unclead** [comment]
Hmmm i checked it with select2 and didn't get any errors

```php
            'name'  => 'user_id',
            'type' => Select2::className(),
            'options' => [
                'data' => [
                    31 => 'item 31',
                    32 => 'item 32',
                    33 => 'item 33',
                    34 => 'item 34',
                    35 => 'item 35',
                    36 => 'item 36',
                ],
                'options' => ['placeholder' => 'Select a state ...'],
                'pluginOptions' => [
                    'allowClear' => true
                ],
            ],
            'enableError' => true,
            'title' => 'User',
            'defaultValue' => 35,
```
When i am adding a new row the widget sets 35 by default.

Give me an example to reproduce the error.

**Comment by bokodi-dev** [comment]
here is an example with ``TabularInput``:

```
[
    'name' => 'type_id',
    'title' => 'Type',
    'type' => Select2::className(),
    'defaultValue' => 1,
    'options' => [
        'data' => [0 => 'a', 1 => 'b', 2 => 'c'],
        'options' => [
            'placeholder' => Yii::t('module/general', 'Select...')
        ]
    ],
    'enableError' => true
],
```

it's not working

**Comment by unclead** [comment]
fixed via a08491f89dc9255ed1afcdf384847a7265962839
you can use master branch to test it

**Comment by bokodi-dev** [comment]
Seems to work now
